src/plugin/basemap/basemapdescriptor.js

```diff
@@ -45,6 +45,11 @@ goog.inherits(plugin.basemap.BaseMapDescriptor, os.data.LayerSyncDescriptor);
  */
 plugin.basemap.BaseMapDescriptor.prototype.setConfig = function(value) {
   this.origConf_ = value;
+
+  // Layer type override from the base map config
+  if (value['layerType']) {
+    this.setType(value['layerType']);
```

Comment:
The compiler does not track types for properties using bracket notation, so you'll need to cast this as a string to match the param requirement of `setType`.